addons/website/static/src/js/widgets/fullscreen_indication.js

```diff
@@ -0,0 +1,42 @@
+/** @odoo-module alias=website.widgets.FullscreenIndication **/
```

Comment:
The alias system is for compatibility, I think you can just not give a name to the module and import it by file name, I see this for example:
```javascript
import AbstractModel from './abstract_model';
```

Reply:
I think that the alias is needed in this case, as the widget is imported in a file that does not use this new es6 syntax (content/website_root.js)

Reply:
Have a look at the doc https://www.odoo.com/documentation/master/reference/javascript_reference.html#native-javascript-modules before saying it is not possible :wink:  You can use `require('@module/path/to/the/file');`

Reply:
Oups sorry about that... 🤦 changing that !

Reply:
It seems that the `require` function does not allow to import directly the new widget, it returns an object that contains the desired class at the key `Symbol.for("default")`. 
So for now to import the widget, this is the required statement : `const FullscreenIndication = require('@website/js/widgets/fullscreen_indication')[Symbol.for("default")];` but to me it seems that the `require` function should return directly the class and not an object containing the class

Reply:
This is because the new module system is compatible with ES-modules, which allows for both a "default" export and a bunch of non-default exports. If it didn't export default in this manner and just returned the whole thing, you would lose the ability to have non-default exports.

Reply:
Yes, non-default exports work fine, the issue is when you use the default export. I used the fault export as the file defines only the widget and nothing else.
The require function is not able to return both the named and default exports ?

Reply:
this should work:
```js
const { [Symbol.for('default')]: DefaultExport, NamedExport1, NamedExport2 } = require('@module/file.js')
```

Reply:
Thanks, I will export using named exports in the end, as it allows for more flexibility. I was a bit confused because I had  in mind the CommonJS syntax used in NodeJS that allows for
```javascript
module.exports = new DAO()
module.exports.initDAO = initDAO // append other functions as named export

// now you have
let DAO = require('_/helpers/DAO');
// DAO by default is exported class or function
DAO.initDAO()
```
But I will continue to read on transpiling, the new ES6 syntax and the native modules, and CommonJS modules

Reply:
Right, but the commonJS syntax doesn't allow you to have named exports without mutating your default export, so in that sense the ES module syntax is more powerful but it makes interop more difficult.

Reply:
:+1: